src/nvim/tui/tui.c

```diff
@@ -1537,11 +1537,11 @@ static void augment_terminfo(TUIData *data, const char *term,
   bool teraterm = terminfo_is_term_family(term, "teraterm");
   bool putty = terminfo_is_term_family(term, "putty");
   bool screen = terminfo_is_term_family(term, "screen");
+  bool tmux = terminfo_is_term_family(term, "tmux") || !!os_getenv("TMUX");
   bool iterm = terminfo_is_term_family(term, "iterm")
     || terminfo_is_term_family(term, "iTerm.app");
   // None of the following work over SSH; see :help TERM .
   bool iterm_pretending_xterm = xterm && iterm_env;
-  bool tmux_wrap = screen && !!os_getenv("TMUX");
```

Comment:
Just taking the old logic at it's word:

    bool tmux_wrap = screen && !!os_getenv("TMUX");
 
that specifically says "we only want to use `TMUX_WRAP` in the specific case where `TERM=screen-256color` and `$TMUX` is defined". That leads me to believe that we should not use `TMUX_WRAP` if `TERM=tmux-256color`. Was the old logic wrong?